x-pack/plugins/lens/public/embeddable/expression_wrapper.tsx

```diff
@@ -53,9 +54,28 @@ export interface ExpressionWrapperProps {
 interface VisualizationErrorProps {
   errors: ExpressionWrapperProps['errors'];
   canEdit: boolean;
+  onExpressionError: ExpressionWrapperProps['onExpressionError'];
+  searchSessionId?: string;
 }
 
-export function VisualizationErrorPanel({ errors, canEdit }: VisualizationErrorProps) {
+export function VisualizationErrorPanel({
+  errors,
+  canEdit,
+  onExpressionError,
+  searchSessionId,
+}: VisualizationErrorProps) {
+  // use a combination of sessionid + error messages to decide whether to trigger a rerender
+  const rerenderKey = `${searchSessionId || ''}-${errors
+    ?.map(({ longMessage }) => longMessage)
+    .join('-')}`;
+
+  const keyRef = useRef(rerenderKey);
+  // Skip error logging when no search session id is passed
```

Comment:
I don't think this comment is relevant.

`// Skip error logging when no search session id is passed`